### PR TITLE
Support latest CDH v5.x and CDH v6.x (6.2.0 currently)

### PR DESCRIPTION
**Pull request by lhoss**
This PR is based on https://github.com/cloudera/cloudera-playbook/pull/23 that includes work from @maciejkowalczyk9ld

The version to deploy is set through one single variable (with current default to latest 6.2.x):
```
#cdh_version: 5.16.1
cdh_version: 6.2.0
```
* I tested deployment of both versions (on AWS, with some other modifications, that I want to contribute separately)
* Another nice thing: all other related variables (like the the repo_url, the parcel_repo, the scm_prepare_database script) are set through a helper dict `configs_by_version` (also to avoid any if/else logic)

---
Msg to the repo maintainers/committers ( @roczei ?):
* Any chance of getting contributions merged in reasonable time?
  * I'm (currently) very motivated to contribute (and there is lot), and even taking extra time to make nice atomic commits/PRs, but if I see no action here I'll not bother anymore creating upstream PRs but only manage a nice+improved fork 👍 
* I highly suggest adding integration tests, ideally using Ansible Molecule (I might have some time in 2w), that could then be run for free on Travis 🎱 
  * alternatively there's some existing work with *vagrant* (I found in a fork): https://github.com/cloudera/cloudera-playbook/compare/master...cgsyam:test_br

### CODE REVIEW

**Comment by roczei** [comment]
Hi @maciejkowalczyk9ld,

You need to first sign and return an ICLA and CCLA before we can accept and redistribute your contribution. Once these are submitted you are free to start contributing to cloudera-playbook. Submit these to CLA@cloudera.com

Here you can find the pdf files:

https://github.com/cloudera/cloudera-playbook/tree/master/icla

Please update this issue (26) if you receive the confirmation from our CLA team.

Thank you!

**Comment by lhoss** [comment]
Hi @roczei 
- Did you see my signed ICLA/CCLA were ack'ed by Cloudera (Casey V.),?
- In case @maciejkowalczyk9ld is not interested in contributing, I can rewrite the initial code I based my PR on.. just need to know!  (ps If I had known earlier about the need to sign docs, I'ld have re-written directly my own code)
- I'm happy to fix conflicts (due to changes in master) in these days, if my PRs really get merged 
  - I suggest getting it done better earlier than later, not that your (master) branch deviates too much 👍 
  - In general it would be great to know  for each PR/contribution if it will get accepted (of course after testing etc)
  - If all PRs acceptable, it would be easier for me todo in 1 PR (because I've done the conflict resolution already between PRs, when I ff-merged them in our fork), but I'm even ok to merge..re-work PR by PR!

**Comment by roczei** [comment]
Hi Laurent (@lhoss),

Yes, I saw that you have signed these docs. Thank you very much for your efforts and dedication what you provide for improving the cloudera-playbook community!

I would like to clarify two things:

1) The cloudera-playbook git repository is not officially supported by Cloudera, but the authors recommend using it.

2) My support regarding the cloudera-playbook is a best effort work. I do not have dedicated time for this. I have added updates to this repository because I am working on a new Cloudera engineering blog article and there are some files  here what I referenced from the blog. It will be published here: https://blog.cloudera.com This will cover two topics:

a) Dynamic Inventory Script for Cloudera Manager: https://github.com/cloudera/cloudera-playbook#dynamic-inventory-script-for-cloudera-manager

b) SSSD setup with Ansible (applicable for RHEL 7 / CentOS 7): https://github.com/cloudera/cloudera-playbook#sssd-setup-with-ansible-applicable-for-rhel-7--centos-7

I would like to help you, therefore I have contacted somebody inside Cloudera who may help by this work: accepting  and reviewing the pull requests, answering your questions, etc. Hopefully he will contact you soon here.

Best regards,

      Gabor

**Comment by lhoss** [comment]
>  would like to help you, therefore I have contacted somebody inside Cloudera who may help by this work: accepting and reviewing the pull requests,

Thanks Gabor!
The one to help, it's not by chance @alexandruanghel  (the maintainer of  https://github.com/hortonworks/ansible-hortonworks, where I did many contribs also)? Actually, as you don't have much time for this ansible, he could be a great help (from inside Cloudera).
Another solution could be to give me committer permissions (that I can earn' by first providing working (molecule)tests .. ).
ps: I'd recommend to do any new features, in feature branches first.. and only merge to master when the feature is done (and tested, etc)

File: group_vars/scm_server.yml [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
 
 scm_repositories:
-  - http://archive.cloudera.com/cdh5/parcels/5.8.3/
-  - https://archive.cloudera.com/cdh5/parcels/{latest_supported}/
+  - "{{ scm_parcel_repository }}"
 
 scm_products:
   - product: CDH
-    version: 5.8.3-1.cdh5.8.3.p0.2
+    version: "{{ cdh_version }}-1.cdh{{ cdh_version }}.p0.2"
```
Comment (dbeech) [inline group_vars/scm_server.yml]: Unfortunately we can't guarantee a parcel product version will have the `.p0.2` suffix. We need a better way of specifying this, it's not great to have the duplication between `cdh_version` variable and a separate one for parcel product.

File: group_vars/all [hunk 2]
```diff
@@ -7,3 +7,26 @@ ad_domain: "{{ krb5_realm.lower() }}"
 computer_ou: ou=Hosts,ou=morhidi,ou=HadoopClusters,ou=morhidi,dc=ad,dc=sec,dc=cloudera,dc=com
 domain: vpc.cloudera.com
 kdc: w2k8-1.ad.sec.cloudera.com
+
+# Only the latest CDH minor version is available in the official repo
```
Comment (dbeech) [inline group_vars/all:32]: Which official repo does this refer to? All CDH versions can be found in archive.cloudera.com.

**Comment by roczei** [comment]
Hi @lhoss 

Finally we have found the owner for the cloudera-playbook repository. @dbeech is a Clouderan as well. He will help you by accepting and reviewing the pull requests, answering the questions, etc.

Best regards,

       Gabor

**Comment by dbeech** [comment]
Merging into cdh6_support branch to rebase onto recent changes - will fix outstanding issues from there.

**Pull request merged**